src/utils/dom.ts

```diff
@@ -0,0 +1,11 @@
+export function getElementDistanceToTop(element: HTMLElement) {
+  var distanceToTop = 0;
+  var currentElement: HTMLElement | null = element;
+
+  while (currentElement) {
```

Comment:
Is this a good alternative solution? https://developer.mozilla.org/en-US/docs/Web/API/Element/getBoundingClientRect

Reply:
I had tried it, but they are not the same thing, the ClientRect will change when the content scroll